src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -20,9 +24,132 @@ public InstructorCourseDetailsPageData(AccountAttributes account) {
 
     public InstructorAttributes currentInstructor;
     public CourseDetailsBundle courseDetails;
-    public List<StudentAttributes> students;
     public List<InstructorAttributes> instructors;
     public String studentListHtmlTableAsString;
+    public ElementTag giveCommentButton;
+    public ElementTag courseRemindButton;
+    public CourseDetailsStudentsTable studentsTable;
+    
+    public void init(InstructorAttributes currentInstructorParam, CourseDetailsBundle courseDetailsParam, 
+                     List<InstructorAttributes> instructorsParam, List<StudentAttributes> students) {
+        this.currentInstructor = currentInstructorParam;
+        this.courseDetails = courseDetailsParam;
+        this.instructors = instructorsParam;
+        
+        boolean isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        this.giveCommentButton = createButton("<span class=\"glyphicon glyphicon-comment glyphicon-primary\"></span>",
```

Comment:
Split this method into multiple methods.
There's no need to use the `this` keyword when it's not necessary; that only makes it harder to read, use it only when it serves a purpose.
If you're using `*Param` as your variable names, don't combine them with `this.*`.